frontstage/views/passwords.py

```diff
@@ -23,56 +22,35 @@ def forgot_password():
 
     if request.method == 'POST' and form.validate():
         email_address = request.form.get('email_address')
-        post_data = {"email_address": email_address}
+        post_data = {"username": email_address}
 
+        response = api_call('POST', app.config['REQUEST_PASSWORD_CHANGE'], json=post_data)
 
-        token_url = app.config['ONS_TOKEN']
-
-        data = {
-            'grant_type': 'password',
-            'client_id': app.config['RAS_FRONTSTAGE_CLIENT_ID'],
-            'client_secret': app.config['RAS_FRONTSTAGE_CLIENT_SECRET'],
-            'username': email_address,
-            'password': 'password'
-        }
-        headers = {
-            'Accept': 'application/json',
-            'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
-        }
-
-        oauth2_response = requests.post(url=token_url, data=data,
-                                        headers=headers, auth=(app.config['RAS_FRONTSTAGE_CLIENT_ID'],
-                                                               app.config['RAS_FRONTSTAGE_CLIENT_SECRET']))
-        # Check to see that this user account is not locked
-        if oauth2_response.status_code == 401:
-            oauth2Error = json.loads(oauth2_response.text)
-            if 'Unauthorized user credentials' in oauth2Error['detail']:
-                logger.warning('Email address is not registered')
+        if response.status_code == 401:
+            error_json = json.loads(response.text).get('error')
+            error_message = error_json.get('data', {}).get('detail')
+            if 'Unauthorized user credentials' in error_message:
+                logger.info('Requesting password change for unregistered email on OAuth2 server')
```

Comment:
Instead of having an explicit line of log we could just log the error_message. Then you'd have something like:
```python
if response.status_code == 401:
    error_json = json.loads(response.text).get('error')
    error_message = error_json.get('data', {}).get('detail')
    logger.info(error_message=error_message)
    if 'Unauthorized user credentials' in error_message:
        template_data = {"error": {"type": {"Email address is not registered"}}}
        return render_template('passwords/forgot-password.html', _theme='default', form=form, data=template_data)

    return render_template('passwords/reset-password.trouble.html', _theme='default',
                           data={"error": {"type": "failed"}})
```